config.go

```diff
@@ -115,3 +109,9 @@ func (c *Config) Write(path string) error {
 
 	return ioutil.WriteFile(path, data, os.ModePerm)
 }
+
+/* unexported */
+func parseConfig(cfg []byte) (spec *ConfigSpec, err error) {
```

Comment:
nit: i personally prefer not having named return types but i'll leave it to @ThePrimeagen to make the last call on it.